frontend/src/app/search/search.component.ts

```diff
@@ -84,7 +85,10 @@ export class SearchComponent extends ParamDirective {
     async initialize(): Promise<void> {
         this.user = await this.authService.getCurrentUserPromise();
         this.corpusSubscription = this.corpusService.currentCorpus
-            .pipe(filter((corpus) => !!corpus))
+            .pipe(
+                filter((corpus) => !!corpus),
+                distinct(corpus => corpus.name),
```

Comment:
Do we need the filter operator above? Shouldn't the `distinct` operator filter enough here?